### PR TITLE
docs and fix loading data in retriever only

### PR DESCRIPTION
**Pull request by Ben-Epstein**
retriever only got mesed up on a merge with main at some point. Now it supports CSV, actual dataset, and dataset directory (like in rage2e)

tested e2e

### CODE REVIEW

File: dalm/training/retriever_only/train_retriever_only.py [hunk 1]
```diff
@@ -226,11 +226,24 @@ def train_retriever(
...
-    dataset = datasets.load_dataset(
-        "csv", data_files={"train": train_dataset_or_csv_path, "validation": test_dataset_or_csv_path}
+    train_dataset = (
+        train_dataset_or_csv_path
+        if isinstance(train_dataset_or_csv_path, Dataset)
+        else datasets.load_from_disk(train_dataset_or_csv_path)
+        if os.path.isdir(train_dataset_or_csv_path)
+        else datasets.load_dataset("csv", data_files=train_dataset_or_csv_path)["train"]
     )
+    data = {"train": train_dataset}
```
Comment (shamanez) [inline dalm/training/retriever_only/train_retriever_only.py:236]: remove this

**Review by shamanez — CHANGES_REQUESTED** [review]

**Review by shamanez — CHANGES_REQUESTED** [review]

File: dalm/training/retriever_only/train_retriever_only.py [hunk 2]
```diff
@@ -226,11 +226,24 @@ def train_retriever(
...
+    if test_dataset_or_csv_path is not None:
```
[the remark this answers is not in the document]
Comment (Ben-Epstein) [reply dalm/training/retriever_only/train_retriever_only.py:237]: Ah okay. It was in the argparse script originally so i kept it, but I'll remove it and rename the command to match rage2e. I was wondering why they didn't match

[the remark this answers is not in the document]
Comment (Ben-Epstein) [reply dalm/training/retriever_only/train_retriever_only.py]: done

File: dalm/training/retriever_only/train_retriever_only.py (hunk 1 above)
Comment (Ben-Epstein) [reply dalm/training/retriever_only/train_retriever_only.py]: yup, updated

**Pull request merged**